pkg/apis/pipeline/v1alpha1/resource_types.go

```diff
@@ -130,15 +130,29 @@ type PipelineResourceList struct {
 	Items           []PipelineResource `json:"items"`
 }
 
-// ResourceFromType returns a PipelineResourceInterface from a PipelineResource's type.
-func ResourceFromType(r *PipelineResource) (PipelineResourceInterface, error) {
+// InputResourceFromType returns a PipelineResourceInterface from a PipelineResource's type.
+func InputResourceFromType(r *PipelineResource) (PipelineResourceInterface, error) {
 	switch r.Spec.Type {
 	case PipelineResourceTypeGit:
 		return NewGitResource(r)
-	case PipelineResourceTypeImage:
-		return NewImageResource(r)
 	case PipelineResourceTypeCluster:
 		return NewClusterResource(r)
+	default:
+		return InputOutputResourceFromType(r)
```

Comment:
i found it confusing that the default for this function calls `InputOutputResourceFromType` - id find it a bit easier to see explicitly which types are allowed for input and output, and always default to an error